src/server/rdb_load.cc

```diff
@@ -1902,6 +1915,32 @@ error_code RdbLoaderBase::HandleCompressedBlobFinish() {
   return kOk;
 }
 
+error_code RdbLoaderBase::HandleJournalBlob(Service* service, DbIndex dbid) {
+  // Read the number of entries in the journal blob.
+  size_t num_entries;
+  bool _encoded;
+  SET_OR_RETURN(LoadLen(&_encoded), num_entries);
+
+  // Read the journal blob.
+  string journal_blob;
+  SET_OR_RETURN(FetchGenericString(), journal_blob);
+
+  io::BytesSource bs{io::Buffer(journal_blob)};
+  JournalReader rd{&bs, dbid};
+
+  // Parse and exectue in loop.
+  size_t done = 0;
+  JournalExecutor ex{service};
+  while (done < num_entries) {
+    journal::ParsedEntry entry{};
+    SET_OR_RETURN(rd.ReadEntry(), entry);
+    ex.Execute(std::move(entry));
+    done++;
+  }
```

Comment:
Once we merge parsing utils, we can add an explicit EOF error. This way we don't need to know the number of entries.